format examples

ranocha opened the pull request:
It would be nice if several people could review this PR, e.g., @trixi-framework/developers 

## TODO

- [x] update style guide

JoshuaLampert commented on examples/dgmulti_2d/elixir_navierstokes_convergence.jl:
```diff
@@ -10,180 +10,226 @@ mu() = 0.01
+             (1.0 - exp(-A * (y - 1.0))) / (y + 2.0)) * cos(pi_t)
+    v1_yy = (sin(pi_x) *
+             (2.0 * A * exp(-A * (y - 1.0)) / (y + 2.0)
+              -
```
These minus and plus signs in separate lines look weird and make this section overly long.

sloede replied:
I found an imho acceptable version in a5d572aa. Could you maybe apply something similar to the other files you found (probably all navierstokes_convergence files for any mesh type)

JoshuaLampert commented on examples/dgmulti_2d/elixir_navierstokes_convergence_curved.jl:
```diff
@@ -36,161 +38,206 @@ mesh = DGMultiMesh(dg, cells_per_dimension, mapping; periodicity=(true, false),
+           + rho * v1_y * v2
+           + rho * v1 * v2_y
+           # stress tensor from x-direction
+           -
```
Here again minus and plus signs in separate lines.

ranocha replied:
It looks like JuliaFormatter is confused by the combination of `+` and `-`. Everything is fine if I just use `+` everywhere. I think we should file an issue

sloede replied:
https://github.com/domluna/JuliaFormatter.jl/issues/739

SimonCan commented on examples/dgmulti_2d/elixir_mhd_reflective_wall.jl:
```diff
@@ -8,70 +8,73 @@ using LinearAlgebra: norm, dot # for use in the MHD boundary condition
+                                               t,
+                                               surface_flux_function,
+                                               equations::IdealGlmMhdEquations2D)
 
```
In function `boundary_condition_velocity_slip_wall` would we rather want the arguments `t` and `surface_flux_function` in different lines?

jlchan replied:
It looks odd to me, I'd prefer `x, t` on the same line.

sloede replied:
It's fixable, e.g., see 6b775b37. Can you maybe directly fix this wherever you find it in this PR?

SimonCan commented on examples/dgmulti_3d/elixir_navierstokes_convergence.jl:
More line breaking plus and minus signs.

SimonCan commented on examples/dgmulti_3d/elixir_navierstokes_convergence.jl:
Could the arguments `x` and `t` not be in the same line?

SimonCan commented on examples/dgmulti_3d/elixir_navierstokes_convergence_curved.jl:
Line breaking plus and minus signs.

jlchan replied:
Can you expand on this? I don't see the line break or `+`/`-` signs you're referring to.

SimonCan replied:
From line 86 on.

SimonCan commented on examples/dgmulti_3d/elixir_navierstokes_convergence_curved.jl:
Should we put `x` and `t` in the same line?

jlchan commented on examples/dgmulti_2d/elixir_advection_diffusion.jl:
```diff
@@ -11,48 +11,52 @@ initial_condition_zero(x, t, equations::LinearScalarAdvectionEquation2D) = SVect
 
 # BC types
-boundary_condition_left = BoundaryConditionDirichlet((x, t, equations) -> SVector(1 + 0.1 * x[2]))
+boundary_condition_left = BoundaryConditionDirichlet((x, t, equations) -> SVector(1 +
+                                                                                  0.1 *
+                                                                                  x[2]))
```
This looks odd having 3 extra line breaks for `SVector`

ranocha replied:
It's too long if we tried to put it in one line. We would have to use shorter variable names if we wanted to achieve that.

jlchan commented on examples/dgmulti_2d/elixir_euler_triangulate_pkg_mesh.jl:
```diff
@@ -13,11 +13,11 @@ meshIO = StartUpDG.triangulate_domain(StartUpDG.RectangularDomainWithHole())
 boundary_condition_convergence_test = BoundaryConditionDirichlet(initial_condition)
 boundary_conditions = (; :outer_boundary => boundary_condition_convergence_test,
-                         :inner_boundary => boundary_condition_convergence_test)
+                       :inner_boundary => boundary_condition_convergence_test)
```
Would be nice to align the symbols if possible?

sloede replied:
@ranocha complained about this before; sadly, it does not seem to be possible 😢

jlchan reviewed (commented):
Looked over all of the `dgmulti_1d` and `2d` elixirs. A few minor comments but nothing too major.

sloede reviewed (commented):
I've randomly looked at a number of files from different mesh types. Mostly, it seems like everything is as good or as bad as before, sometimes there are slight improvements

In some cases (already noted by @JoshuaLampert @SimonCan @jlchan), there exists a somewhat weird formatting with binary operators being alone on one line. This is obviously a decline in readability. However, they are fixable (by moving the binary op to the previous line). Thus, from my side there are no hard reasons to not proceed with merging this once everyone had.a chance to review (and improve) things

JoshuaLampert commented on examples/dgmulti_2d/elixir_navierstokes_convergence.jl:
```diff
@@ -10,180 +10,192 @@ mu() = 0.01
+           + rho * v1 * v2_y -
+           # stress tensor from x-direction
+           4.0 / 3.0 * v1_xx * mu_ +
+           2.0 / 3.0 * v2_xy * mu_ -
```
It is still a bit weird that sometimes the binary operator is in the beginning of the line and sometimes at the end, but I could live with that.

ranocha commented:
I updated this PR based on the recent `main`. I also added the workaround for the weird formatting of binary operators everywhere. Some stochastic last reviews would be nice before merging this when tests pass.

There are some instances where the arguments `x, t` are on separate lines when defining boundary conditions - but that's necessary given our line length limit we use right now. The other cases with `t,` alone on a separate line are fixed.

sloede reviewed (approved):
I did some spot checks but nothing rang besides that, like for the `test/` folder, this reformatting is unfortunately not a clear improvement for readability but rather a lateral move. I still think it is useful to do!

It would be good if maybe @JoshuaLampert or @lchristm could have a look as well, maybe also @DanielDoehring? Not at everything, of course, but at least at a few random elixirs...

DanielDoehring commented:
Looks alright to me, we loose some "nice" formatting like 

```julia
coordinates_min = (-1.0, -1.0) # minimum coordinates (min(x), min(y))
coordinates_max = ( 1.0,  1.0) # maximum coordinates (max(x), max(y))
```

which becomes


```julia
coordinates_min = (-1.0, -1.0) # minimum coordinates (min(x), min(y))
coordinates_max = (1.0, 1.0) # maximum coordinates (max(x), max(y))
```

but apart from that this looks fine to me.

JoshuaLampert commented:
> Looks alright to me, we loose some "nice" formatting like
> 
> ```julia
> coordinates_min = (-1.0, -1.0) # minimum coordinates (min(x), min(y))
> coordinates_max = ( 1.0,  1.0) # maximum coordinates (max(x), max(y))
> ```
> 
> which becomes
> 
> ```julia
> coordinates_min = (-1.0, -1.0) # minimum coordinates (min(x), min(y))
> coordinates_max = (1.0, 1.0) # maximum coordinates (max(x), max(y))
> ```
> 
> but apart from that this looks fine to me.

Personally, I actually  prefer the formatted one.

JoshuaLampert commented:
Does someone know why some of the tests fail? [Some of them](https://github.com/trixi-framework/Trixi.jl/actions/runs/6730714984/job/18293950702?pr=1531#step:7:4680) even quite considerably (the actual error differs by one order of magnitude to the expected one)... We having similar issues also in other PRs, but strangely not the same tests fail.

ranocha commented:
> Does someone know why some of the tests fail? [Some of them](https://github.com/trixi-framework/Trixi.jl/actions/runs/6730714984/job/18293950702?pr=1531#step:7:4680) even quite considerably (the actual error differs by one order of magnitude to the expected one)... We having similar issues also in other PRs, but strangely not the same tests fail.

I have no clue... 😞

ranocha commented:
Since CI also fails in other PRs, I think we should go ahead and merge this PR to avoid merge conflicts with other works. Okay for you, @sloede?

sloede commented:
> Since CI also fails in other PRs, I think we should go ahead and merge this PR to avoid merge conflicts with other works. Okay for you, @sloede?

I think we should wait until all tests have run, but yes, I agree that if this is just the same errors as elsewhere, we can merge

DanielDoehring commented:
> Does someone know why some of the tests fail? [Some of them](https://github.com/trixi-framework/Trixi.jl/actions/runs/6730714984/job/18293950702?pr=1531#step:7:4680) even quite considerably (the actual error differs by one order of magnitude to the expected one)... We having similar issues also in other PRs, but strangely not the same tests fail.

So for the elixirs using the error-based timestepping I remember that there where actually different errors depending on the machine. This time, I can actually reproduce the errors on my machine and we have differences for other examples using CFL-based timestepping. 

In principle, we should be able to find the point where things went wrong, right...?

ranocha commented:
Yes, you're right. Did some update of one of the time stepping dependencies introduce the change we see here?

jlchan commented on examples/tree_2d_dgsem/elixir_navierstokes_convergence.jl:
```diff
@@ -8,163 +8,170 @@ prandtl_number() = 0.72
+           2.0 / 3.0 * v1_xy -
+           4.0 / 3.0 * v2_y * v2_y * mu_ +
+           2.0 / 3.0 * v1_x * v2_y * mu_
+           -
+           -
```
Formatting appears to have removed something crucial

JoshuaLampert replied:
I don't think the formatter did that, but the manual formatting, e.g., [here](https://github.com/trixi-framework/Trixi.jl/pull/1531/commits/78831ef82beb287042102eb272295b7b6e527bf7?diff=unified&w=0#diff-aefb6b57a33d9112f7244d28e7fb1b7ee8d40e2076e1c7e7570e65681381486fR163).
